Add lint rule to prevent server code being imported into client

joshdover opened the pull request:
## Summary

Fixes #49053

This adds new configurations for the `no-restricted-paths` eslint rule to prevent:
- Importing any server-side modules into any modules that are not in a `server/` directory.
  - This includes `common` directories often used by plugins. Necessary to ensure that modules containing shared code do not import anything that should be exclusively used on the server.
  - This helps avoid issues where server code inadvertently ends up in a Webpack bundle that breaks the front-end. When these problems happen it is often hard to understand why, and this rule should help.
- Importing any code from `ui/*` or `uiExports/*` into New Platform plugins.
  - These modules will be deleted in the future and should not be supported in New Platform plugins.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

joshdover committed:
Add lint rule to prevent server code being imported into client

joshdover commented:
Didn't get this rule quite right, but now that I have some failures I can play around with it. Will circle back next week.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/)
* Commit: ea9488253be526a37787c489a59da432bd28344e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/","number":"14031","commit":"ea9488253be526a37787c489a59da432bd28344e"}]}
PIPELINE-->

restrry commented on .eslintrc.js:
```diff
@@ -345,6 +345,16 @@ module.exports = {
+                  'x-pack/plugins/server/**/*',
+                ],
+                from: ['src/plugins/*/server/**/*', 'x-pack/plugins/*/server/**/*'],
+                errorMessage: 'Server code cannot be imported into client code',
```
we need to add  `allowSameFolder: true`

pgayvallet commented:
Fixes https://github.com/elastic/kibana/issues/49053 btw

restrry commented on .eslintrc.js:
```diff
@@ -345,6 +345,16 @@ module.exports = {
+              {
+                target: [
+                  'src/plugins/**/*',
+                  '!src/plugins/server/**/*',
```
`!src/plugins/*/server/**/*`?

joshdover replied:
Ah, yes :)

restrry commented on .eslintrc.js:
```diff
@@ -345,6 +345,16 @@ module.exports = {
+                  'src/plugins/**/*',
+                  '!src/plugins/server/**/*',
+                  'x-pack/plugins/**/*',
+                  'x-pack/plugins/server/**/*',
```
`!x-pack/plugins/*/server/**/*` ?

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/)
* Commit: 3e1bd47443c8ea0bebbf1538334966d80ee9ac9b

### History
* :broken_heart: [Build #14031](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/) failed ea9488253be526a37787c489a59da432bd28344e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/","number":"14031","commit":"ea9488253be526a37787c489a59da432bd28344e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/","number":"14267","commit":"3e1bd47443c8ea0bebbf1538334966d80ee9ac9b"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-platform (Team:Platform)

kobelb commented on x-pack/plugins/security/server/authentication/authenticator.ts:
```diff
@@ -31,7 +31,7 @@ import {
 import { DeauthenticationResult } from './deauthentication_result';
 import { Tokens } from './tokens';
-import { SessionInfo } from '../../public/types';
+import { SessionInfo } from '../../public';
```
FWIW, I like the changes you made. Did the lint rule require these changes?

joshdover replied:
Yes it does. If you don't want this to be part of the public contract, we can move this up into a shared `types` file that only security can import from.

restrry replied:
BTW shouldn't we ban public code to be imported from server as well? I'm concern about cases when a plugin accesses browser specific API and can throw on the server side

joshdover replied:
I'm not against that, was just focusing this change on fixing the Webpack bundling problems. It looks like there are more violations of this pattern, let me see about doing that change in a separate PR.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/)
* Commit: ab8732570e10d08ef12c1a10266600e61c61a740

### History
* :broken_heart: [Build #14267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/) failed 3e1bd47443c8ea0bebbf1538334966d80ee9ac9b
* :broken_heart: [Build #14031](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/) failed ea9488253be526a37787c489a59da432bd28344e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/","number":"14031","commit":"ea9488253be526a37787c489a59da432bd28344e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/","number":"14267","commit":"3e1bd47443c8ea0bebbf1538334966d80ee9ac9b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/","number":"14315","commit":"ab8732570e10d08ef12c1a10266600e61c61a740"}]}
PIPELINE-->

restrry commented on .eslintrc.js:
```diff
@@ -337,13 +337,20 @@ module.exports = {
+                errorMessage: 'Plugins may only import from top-level public and server modules.',
+              },
+              {
+                target: ['(src|x-pack)/plugins/**/*', '!(src|x-pack)/plugins/*/server/**/*'],
```
Shouldn't it complain when a plugin reads from server/types? Probably we have it blacklisted, I'm from phone and cannot see

joshdover replied:
Are you saying we should allow imports from `server/types` or not? In the current state of the PR it does not allow this import. I think we should instead, encourage plugins to move shared things into a `common` directory just like other code.

restrry replied:
agree

joshdover committed:
Fix bad imports

smith reviewed (approved):
APM stuff looks good!

streamich reviewed (commented):
Shall we also add rules that:

- `/src/plugins/*/server/**/*` should not be able to import from `/src/plugins/*/public/**/*`
- `/src/plugins/*/public/**/*` should not be able to import from `/src/plugins/*/server/**/*`
- `/src/plugins/*/common/**/*` should not be able to import from server, nor public

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/)
* Commit: a0f830fed1abf9346f095e4e7535d8ed7fe900d0

### History
* :broken_heart: [Build #14315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/) failed ab8732570e10d08ef12c1a10266600e61c61a740
* :broken_heart: [Build #14267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/) failed 3e1bd47443c8ea0bebbf1538334966d80ee9ac9b
* :broken_heart: [Build #14031](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/) failed ea9488253be526a37787c489a59da432bd28344e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/","number":"14031","commit":"ea9488253be526a37787c489a59da432bd28344e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/","number":"14267","commit":"3e1bd47443c8ea0bebbf1538334966d80ee9ac9b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/","number":"14315","commit":"ab8732570e10d08ef12c1a10266600e61c61a740"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/","number":"14356","commit":"a0f830fed1abf9346f095e4e7535d8ed7fe900d0"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/)
* Commit: 2a10c37d8dee2ac237ded4e95c0656202b70037b

### History
* :broken_heart: [Build #14356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/) failed a0f830fed1abf9346f095e4e7535d8ed7fe900d0
* :broken_heart: [Build #14315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/) failed ab8732570e10d08ef12c1a10266600e61c61a740
* :broken_heart: [Build #14267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/) failed 3e1bd47443c8ea0bebbf1538334966d80ee9ac9b
* :broken_heart: [Build #14031](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/) failed ea9488253be526a37787c489a59da432bd28344e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/","number":"14031","commit":"ea9488253be526a37787c489a59da432bd28344e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/","number":"14267","commit":"3e1bd47443c8ea0bebbf1538334966d80ee9ac9b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/","number":"14315","commit":"ab8732570e10d08ef12c1a10266600e61c61a740"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/","number":"14356","commit":"a0f830fed1abf9346f095e4e7535d8ed7fe900d0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/","number":"14372","commit":"2a10c37d8dee2ac237ded4e95c0656202b70037b"}]}
PIPELINE-->

joshdover commented:
> Shall we also add rules that:
> 
> * `/src/plugins/*/server/**/*` should not be able to import from `/src/plugins/*/public/**/*`

As mentioned above, I will do this in a separate change as there are more things to fix here and this is less urgent.

> * `/src/plugins/*/public/**/*` should not be able to import from `/src/plugins/*/server/**/*`
> * `/src/plugins/*/common/**/*` should not be able to import from server, nor public

Both of these are already covered in this PR. Nothing may import from `server` that is not also a `server` directory. This eliminates the immediate problem where Webpack bundles get broken.

streamich commented:
I see, thanks, every time I get confused by `from` and `target` meaning in that ESLint rule.

jfsiii commented:
Is this only for NP plugins? It seems useful for many. Anecdotally, I've done this twice in the last few months in an x-pack/legacy plugin and would love some protection from myself :)

Just curious, nothing to hold up this PR.

lukeelmers commented:
> Is this only for NP plugins? It seems useful for many.

++ This is what I was wondering.

Would be great if the imports-from-server part could be run on legacy plugins too. For example, these updates would not have prevented [this regression](https://github.com/elastic/kibana/pull/52248) which was fixed recently, because it was in `x-pack/legacy/plugins`.

Of course, `ui/` and `uiExports/` should probably still be allowed in the legacy world, but is there any reason why we wouldn't want to enforce the server imports bit?

afgomez reviewed (approved):
Infra changes LGTM

joshdover committed:
Add server imports check to legacy plugins

joshdover committed:
Fix bad legacy imports

joshdover commented:
Added a rule to cover legacy plugins as well. There are a lot more violations here, some of which were trivial to fix, others which are not. For the more complex ones I have added eslint-ignore lines. These should be cleaned up when migrating to the New Platform.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/)
* Commit: ca8e5fddc5b585739b271ad39996919a214e6d81

### History
* :green_heart: [Build #14372](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/) succeeded 2a10c37d8dee2ac237ded4e95c0656202b70037b
* :broken_heart: [Build #14356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/) failed a0f830fed1abf9346f095e4e7535d8ed7fe900d0
* :broken_heart: [Build #14315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/) failed ab8732570e10d08ef12c1a10266600e61c61a740
* :broken_heart: [Build #14267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/) failed 3e1bd47443c8ea0bebbf1538334966d80ee9ac9b
* :broken_heart: [Build #14031](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/) failed ea9488253be526a37787c489a59da432bd28344e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14031/","number":"14031","commit":"ea9488253be526a37787c489a59da432bd28344e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/","number":"14267","commit":"3e1bd47443c8ea0bebbf1538334966d80ee9ac9b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/","number":"14315","commit":"ab8732570e10d08ef12c1a10266600e61c61a740"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/","number":"14356","commit":"a0f830fed1abf9346f095e4e7535d8ed7fe900d0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/","number":"14372","commit":"2a10c37d8dee2ac237ded4e95c0656202b70037b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/","number":"14972","commit":"ca8e5fddc5b585739b271ad39996919a214e6d81"}]}
PIPELINE-->

mikecote reviewed (approved):
Stack services changes LGTM 👍 Created #52896 for us to change Task Manager's folder structure.

walterra reviewed (approved):
ML changes LGTM

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/)
* Commit: 7015c366911bcdf046b53f699da4f5878f8a7ae0

### History
* :broken_heart: [Build #14972](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/) failed ca8e5fddc5b585739b271ad39996919a214e6d81
* :green_heart: [Build #14372](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/) succeeded 2a10c37d8dee2ac237ded4e95c0656202b70037b
* :broken_heart: [Build #14356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/) failed a0f830fed1abf9346f095e4e7535d8ed7fe900d0
* :broken_heart: [Build #14315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/) failed ab8732570e10d08ef12c1a10266600e61c61a740
* :broken_heart: [Build #14267](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/) failed 3e1bd47443c8ea0bebbf1538334966d80ee9ac9b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14267/","number":"14267","commit":"3e1bd47443c8ea0bebbf1538334966d80ee9ac9b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/","number":"14315","commit":"ab8732570e10d08ef12c1a10266600e61c61a740"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/","number":"14356","commit":"a0f830fed1abf9346f095e4e7535d8ed7fe900d0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/","number":"14372","commit":"2a10c37d8dee2ac237ded4e95c0656202b70037b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/","number":"14972","commit":"ca8e5fddc5b585739b271ad39996919a214e6d81"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/","number":"15188","commit":"7015c366911bcdf046b53f699da4f5878f8a7ae0"}]}
PIPELINE-->

joshdover committed:
Fix typings

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/)
* Commit: a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a

### History
* :broken_heart: [Build #15188](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/) failed 7015c366911bcdf046b53f699da4f5878f8a7ae0
* :broken_heart: [Build #14972](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/) failed ca8e5fddc5b585739b271ad39996919a214e6d81
* :green_heart: [Build #14372](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/) succeeded 2a10c37d8dee2ac237ded4e95c0656202b70037b
* :broken_heart: [Build #14356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/) failed a0f830fed1abf9346f095e4e7535d8ed7fe900d0
* :broken_heart: [Build #14315](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/) failed ab8732570e10d08ef12c1a10266600e61c61a740

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14315/","number":"14315","commit":"ab8732570e10d08ef12c1a10266600e61c61a740"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/","number":"14356","commit":"a0f830fed1abf9346f095e4e7535d8ed7fe900d0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/","number":"14372","commit":"2a10c37d8dee2ac237ded4e95c0656202b70037b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/","number":"14972","commit":"ca8e5fddc5b585739b271ad39996919a214e6d81"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/","number":"15188","commit":"7015c366911bcdf046b53f699da4f5878f8a7ae0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/","number":"15220","commit":"a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15237/)
* Commit: f25494dd792ecca8262cf2d68cca580c0eb7f9cd

### History
* :broken_heart: [Build #15220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/) failed a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a
* :broken_heart: [Build #15188](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/) failed 7015c366911bcdf046b53f699da4f5878f8a7ae0
* :broken_heart: [Build #14972](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/) failed ca8e5fddc5b585739b271ad39996919a214e6d81
* :green_heart: [Build #14372](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/) succeeded 2a10c37d8dee2ac237ded4e95c0656202b70037b
* :broken_heart: [Build #14356](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/) failed a0f830fed1abf9346f095e4e7535d8ed7fe900d0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14356/","number":"14356","commit":"a0f830fed1abf9346f095e4e7535d8ed7fe900d0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/","number":"14372","commit":"2a10c37d8dee2ac237ded4e95c0656202b70037b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/","number":"14972","commit":"ca8e5fddc5b585739b271ad39996919a214e6d81"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/","number":"15188","commit":"7015c366911bcdf046b53f699da4f5878f8a7ae0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/","number":"15220","commit":"a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15237/","number":"15237","commit":"f25494dd792ecca8262cf2d68cca580c0eb7f9cd"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15277/)
* Commit: a9816d5378686c67f8308a1cfd4822242615ff72

### History
* :green_heart: [Build #15237](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15237/) succeeded f25494dd792ecca8262cf2d68cca580c0eb7f9cd
* :broken_heart: [Build #15220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/) failed a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a
* :broken_heart: [Build #15188](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/) failed 7015c366911bcdf046b53f699da4f5878f8a7ae0
* :broken_heart: [Build #14972](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/) failed ca8e5fddc5b585739b271ad39996919a214e6d81
* :green_heart: [Build #14372](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/) succeeded 2a10c37d8dee2ac237ded4e95c0656202b70037b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14372/","number":"14372","commit":"2a10c37d8dee2ac237ded4e95c0656202b70037b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/","number":"14972","commit":"ca8e5fddc5b585739b271ad39996919a214e6d81"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/","number":"15188","commit":"7015c366911bcdf046b53f699da4f5878f8a7ae0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/","number":"15220","commit":"a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15237/","number":"15237","commit":"f25494dd792ecca8262cf2d68cca580c0eb7f9cd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15277/","number":"15277","commit":"a9816d5378686c67f8308a1cfd4822242615ff72"}]}
PIPELINE-->

lizozom reviewed (approved):
@streamich comments aside, the changes to @elastic/kibana-app-arch files are LGTM

joshdover commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'master' into np/server-client-imports

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15456/)
* Commit: da66f373e41894b2640ead298f55d9f57ff0aa65

### History
* :green_heart: [Build #15277](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15277/) succeeded a9816d5378686c67f8308a1cfd4822242615ff72
* :green_heart: [Build #15237](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15237/) succeeded f25494dd792ecca8262cf2d68cca580c0eb7f9cd
* :broken_heart: [Build #15220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/) failed a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a
* :broken_heart: [Build #15188](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/) failed 7015c366911bcdf046b53f699da4f5878f8a7ae0
* :broken_heart: [Build #14972](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/) failed ca8e5fddc5b585739b271ad39996919a214e6d81

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/14972/","number":"14972","commit":"ca8e5fddc5b585739b271ad39996919a214e6d81"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15188/","number":"15188","commit":"7015c366911bcdf046b53f699da4f5878f8a7ae0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15220/","number":"15220","commit":"a5394fc5cd0d0ffcd949aca1c1b6a1162f0f7c6a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15237/","number":"15237","commit":"f25494dd792ecca8262cf2d68cca580c0eb7f9cd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15277/","number":"15277","commit":"a9816d5378686c67f8308a1cfd4822242615ff72"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15456/","number":"15456","commit":"da66f373e41894b2640ead298f55d9f57ff0aa65"}]}
PIPELINE-->

joshdover commented:
Opened #53021 to track adding a lint rule to forbid importing in the other direction (client code -> server code)
